day1.js

```diff
@@ -0,0 +1,65 @@
+const formContainer = document.getElementById('formContainer');
+const bookTitle = document.getElementById('title');
+const bookAuthor = document.getElementById('author');
+const dynamicBooksDiv = document.getElementById('books-wrapper');
+
+const books = JSON.parse(localStorage.getItem('booksDetails')) || [];
+
+const addBook = (title, author, id) => {
+  books.push({
+    title,
+    author,
+    id,
+  });
+  localStorage.setItem('booksDetails', JSON.stringify(books));
+  return { title, author, id };
+};
+
+const createBook = ({ title, author, id }) => {
+  const div = document.createElement('div');
+  const h3 = document.createElement('h3');
+  const h4 = document.createElement('h4');
+  const removeBtn = document.createElement('button');
+
+  div.className = 'book-div';
+  h3.innerText = `Title : ${title}`;
+  h4.innerText = `Author :${author}`;
+  removeBtn.innerHTML = 'remove';
+  removeBtn.dataset.id = id;
+  removeBtn.type = 'button';
+  removeBtn.className = 'removeBtn';
+
+  div.append(h3, h4, removeBtn);
+  dynamicBooksDiv.appendChild(div);
+};
+
+books.forEach(createBook);
+
+formContainer.onsubmit = (e) => {
+  e.preventDefault();
+  let id = Math.floor((1 + Math.random()) * 0x10000)
+    .toString(16)
+    .substring(1);
+  const newBook = addBook(bookTitle.value, bookAuthor.value, id);
+  createBook(newBook);
+  bookTitle.value = '';
+  bookAuthor.value = '';
+  id = '';
+};
+
+const deleteBook = (id) => {
+  books.filter((book, index) => {
+    if (book.id === id) {
+      return books.splice(index, 1);
+    }
+    return books.splice(index, 1);
+  });
+  localStorage.setItem('booksDetails', JSON.stringify(books));
+};
+
+document.querySelectorAll('.removeBtn').forEach((removeBtn) => {
+  removeBtn.addEventListener('click', (e) => {
+    e.preventDefault();
+    deleteBook(removeBtn.dataset.id);
+  });
+});
```

Comment:
As a result of your refactor, you seem to have introduced new bugs to your app, which probably was not present during the previous review.
- [ ] First of all, deleting book a book from the app proves to be difficult. It appears out of a list of books, only one can be credibly clicked to trigger the **`deleteBook`** method. Regardless of that, the wrong book is often deleted still. Take a look at the image below. I added 3 books and deleted one, but two are deleted from the localStorage. So when I reload the page only one book remains. This is unexpected behavior. Kindly fix it (see my suggestion below).

  Books are added | One book is deleted | Page is reloaded
  :---:|:---:|:---:
  ![nemwel-1st](https://user-images.githubusercontent.com/36542986/153066581-5adb7a65-4e26-4a4e-93de-9050890386f7.jpg) | ![nemwel-2nd](https://user-images.githubusercontent.com/36542986/153066583-8f03ff05-6fa9-4fac-9e19-509c7b6aa9ff.jpg) | ![nemwel-3rd](https://user-images.githubusercontent.com/36542986/153066586-e2bd8e5f-f2c0-42bc-a65d-b67b6ab8edba.jpg)



  **SUGGESTION:**
  I think there is some inconsistency with the **books** array as it gets manipulated which makes it unable to accurately track your books. Since your **`deletebook`** method only updates the localStorage, I would suggest you refactor your deleteBook method to look like this:
  ```
  const deleteBook = (id) => {
    const mybooks = JSON.parse(localStorage.getItem('booksDetails'));
    const filteredBooks = mybooks.filter((book) => book.id !== id)
    localStorage.setItem('booksDetails', JSON.stringify(filteredBooks));
  };
  ```
  This way, at every point this method works with the latest localStorage details.

  Now to implement the actual action trigger, you do not need line 60 - 65 you can delete them. You can attach your event listener directly to the remove button upon creation in **`createBook`**. Back to your **`createBook`** method add this line to towards the end of the method:

  ```
  removeBtn.addEventListener('click', () => {
    deleteBook(id)
    dynamicBooksDiv.removeChild(div)
  })
  ```

Reply:
Thank you @TedLivist  for this review.
I have added your suggested changes and everything now works perfectly well.